### DISCUSSION TITLE
Wordpress to Quarto migration: URL redirect?

### DISCUSSION THREAD

**Discussion by vinhdizzo** [quarto-dev/quarto-cli#2953] [Q&A] [answered]
Hi there,

I'm looking to migrate a wordpress blog to one based on Quarto.  My posts in wordpress have URL's of the form `https://blog.domain.com/blog/yyyy/mm/dd/title`.  I don't mind migrating to a different structure (eg, defaults in quarto).  For links embedded in the entries, I feel I could write a shell script to modify all the links in the markdown files to the new URL's.  However, what I'd like to keep intact are the usability of the original links as other people's blog posts may reference those old URL's.  I'm thinking the cleanest approach is to set up URL re-direct on the 350 or so original links, redirecting to the latest URL's.  What this does effectively is to not turn my original 350 posts/links into dead links.

Wanted to check in if you all believe this is the best/simplest approach, or is there something easier that I'm not thinking of?

Thank you!

**Comment by mcanouil** [comment]
Hi,

If you did not, I suggest you try the `aliases` option described in <https://quarto.org/docs/websites/website-navigation.html#redirects> which should be the easiest approach.

**Comment by vinhdizzo** [comment]
Thanks @mcanouil , I did see that section on `aliases`.  The section also noted relying on your web server's functionality, which is why I asked if the nginx URL redirect is preferred.  Just not sure which is the recommended path.

Regarding `aliases`, can I use full URL's like `https://blog.mydomain.com/blog/yyyy/mm/dd/title`  in the field?  The examples just show relative paths to pages.

Thanks again.

**Reply by mcanouil** [reply]
I think the main concern about the method to redirect is performance which I don't know anything about.

Regarding the aliases, they are relative to the apex/domain, not to the page, thus you would use `blog/yyyy/mm/dd/title`

**Comment by vinhdizzo** [comment]
Thanks for clarifying `aliases`.  So I guess if we migrate from `blog.mydomain.com` to `www.mydomain.com/blog`, then the `aliases` approach wouldn't work.  I'll go with the URL redirect option then.

**Reply by mcanouil** [reply]
I am not entirely sure because I do not know how internally aliases are used, but switching from a subdomain to a subdirectory of an apex seems to be something you would do using CNAME, .htaccess, etc.

**Reply by vinhdizzo** [reply]
Sorry for the confusion.  I am changing both domain/subdomain and the path of the URL's.

**Comment by mcanouil** [comment]
In your case I would go for 301 redirections.
This will allow you to replace your old URLs by the new ones in search engines, while still allowing old URLs to work.

**Comment by mcanouil** [comment, marked as answer]
Can someone from the Quarto team give a bit more information about how aliases work in Quarto? 🤔
It might be useful to know what kind of redirection it is to help the users decide whether or not, the aliases is the way to go.

**Reply by dragonstyle** [reply]
Aliases actually generate a html page with a script that updates the `window.location` (since we don't know the deployment context, we can't respond to HTTP requests directly with a `301` or `302` or what not). I think aliases make sense when you're shifting page and don't need a centralized list of the redirects, though I do generally think if you know the deployment context and that context has some native support for redirects (especially if using native HTTP redirects), that seems preferable to me.

**Reply by vinhdizzo** [reply]
Thanks @dragonstyle for clarifying. This confirms that for my case, URL redirect from the webserver side is preferred.

**Reply by mcanouil** [reply]
Thanks @dragonstyle
